### PR TITLE
farm: add whitelisting addresses for farm endpoints

### PR DESCRIPTION
**Pull request by claudiulataretu**
- Address whitelisting is used for farm accepting LKMEX as farming token
where proxy will be whitelisted in farm contract

Signed-off-by: Claudiu Ion Lataretu <claudiu.lataretu@gmail.com>

### CODE REVIEW

File: dex/farm/src/lib.rs [hunk 1]
```diff
@@ -210,6 +221,15 @@ pub trait Farm:
         require!(payment_token_id == farm_token_id, "Bad input token");
         require!(amount > 0, "Payment amount cannot be zero");
 
+        let caller = self.blockchain().get_caller();
```
Comment (sasurobert) [inline dex/farm/src/lib.rs:224]: you have to make one function out of lines 224-232. And use a require for it. one liner for this endpoint.

File: dex/farm/src/lib.rs [hunk 2]
```diff
@@ -112,6 +112,8 @@ pub trait Farm:
         &self,
         #[var_args] opt_accept_funds_func: OptionalArg<ManagedBuffer>,
     ) -> SCResult<EnterFarmResultType<Self::Api>> {
+        let apr_multiplier = self.locked_rewards_apr_multiplier().get();
+        require!(apr_multiplier > 1, "farm does not support locked rewards");
```
Comment (sasurobert) [inline dex/farm/src/lib.rs:116]: I think that it is better to make a completely new set of contracts - one which generates locked rewards - thus metaESDTs and one which generates with ESDTs. That will make the overall contracts smaller.

File: dex/farm/src/lib.rs (hunk 1 above)
Comment (sasurobert) [reply dex/farm/src/lib.rs:224]: I would make a separate version for this contract. And would have especially for this case. You can start a new feat/branch - where you say farm with whitelist.

**Review by catalinnnn — CHANGES_REQUESTED** [review]
Looks good, but needs testing, we cannot keep adding changes without adding mandos tests

**Pull request closed without merging**